common/usr/sbin/MSMilter

```diff
@@ -117,9 +117,19 @@ sub connect_callback
 
         if (defined $sockaddr_in)
         {
-            ($port, $iaddr) = sockaddr_in($sockaddr_in);
-            $ip = inet_ntoa($iaddr);
-            $message .= ' [' . $ip . ']';
+            my $family = sockaddr_family($sockaddr_in);
+            if ($family eq AF_INET) {
+                ($port, $iaddr) = sockaddr_in($sockaddr_in);
+                $ip = inet_ntop(AF_INET, $iaddr);
```

Comment:
I tried the code but now I get the error following error

    Undefined subroutine &main::inet_ntop called at /usr/sbin/MSMilter line 127.

Reply:
I added the line

    use Socket qw(inet_ntop);

and now it works

Reply:
I have to revise this. The milter runs, but it reruns again and again. Somehow the block in line 137 doesn't catch.

Reply:
Added.  Apparently Socket doesn't include that one by default.